pulley/src/lib.rs

```diff
@@ -1,6 +1,8 @@
 //! The pulley bytecode for fast interpreters.
 
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![cfg_attr(feature = "tail_calls", feature(explicit_tail_calls))]
+#![cfg_attr(feature = "tail_calls", allow(incomplete_features, unstable_features))]
```

Comment:
As Alex mentioned, to better integrate with our CI infrastructure (which passes `--all-features` to tests), let's replace all `feature = "tail_calls"` in `cfg(...)`s with `cfg(pulley_tail_calls)` and `cfg(not(pulley_tail_calls))`.

Then we can turn it on and off via the `RUSTFLAGS` env var, for example

```
RUSTFLAGS="--cfg pulley_tail_calls" cargo test --all-features
```

And then by default CI will test the regular-calls version, but we can also [add a step to the "Monolith Checks" CI job](https://github.com/bytecodealliance/wasmtime/blob/e38ffa197cf951ed513896fc63bb8e5c7b23135a/.github/workflows/main.yml#L465-L515) that sets the above `RUSTFLAGS` to check that the tail-calls version compiles.